tests/rpmgeneral.at

```diff
@@ -35,6 +35,24 @@ AT_CHECK([runroot rpmbuild --version],[0],
 ])
 AT_CLEANUP
 
+# Check that libtool versioning matches expectations, it's easy to screw up.
+AT_SETUP([rpm library version])
+AT_KEYWORDS([basic])
+AT_CHECK([
+cd ${RPMTEST}/${RPMLIBDIR}
+ls lib*.so.*
+],
+[0],
+[librpm.so.9
+librpm.so.9.1.0
+librpmbuild.so.9
+librpmbuild.so.9.1.0
+librpmio.so.9
+librpmio.so.9.1.0
+librpmsign.so.9
+librpmsign.so.9.1.0
```

Comment:
So is this check only supposed to maintain that the sonames are intended to be frozen?

Reply:
Oh, no. The point is that you have to explicitly update the versions anytime you touch the libtool versioning, which forces you to double-check the intended effect.

Reply:
So then do we have a guide of _how_ we're supposed to fiddle with libtool versioning somewhere?

Reply:
https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html

Following those instructions creates perfectly predictable and sensible versions in the filename, one just needs to remember the libtool version *you* pass is *an abstraction* that libtool then turns into something suitable for the OS. On unix-like systems it's the libraryt's major.minor.micro -style version (with major equaling the soname) we're all familiar with, on some other OS I guess it can be something else (otherwise, why would the abstraction exist?)